acceptance/testdata/secret/secret-repo-env.txtar

```diff
@@ -0,0 +1,28 @@
+# Force GitHub CLI to treat testscript as TTY
+env GH_FORCE_TTY=80
```

Comment:
> Can you talk to me a bit about why you feel you need to have a TTY to have confidence that repo secrets are being interacted with correctly?

The results of `gh secret` appeared to be acting in a non-TTY manner, which resulted in certain commands having no output in certain situations, making it difficult to test.

Reply:
> The results of gh secret appeared to be acting in a non-TTY manner, which resulted in certain commands having no output in certain situations, making it difficult to test.

What's not clear to me is **why** you need to test the output of these commands. What confidence about `gh` working with tenants does checking stdout here give you:

```
exec gh secret set TESTSCRIPTS_ENV --env testscripts --body 'does not matter as cannot read it once set'
stdout 'Set Actions secret TESTSCRIPTS_ENV for'
```

That this doesn't:

```
exec gh secret list --env testscripts
stdout 'TESTSCRIPTS_ENV\s+less than a minute ago'
```

> Could you help me understand how this test doesn't satisfy requirements

It absolutely satisfies the Acceptance Criteria. 

What it also does is introduce a difference in how the scripts are being written because it:
 * Asserts on things that other scripts aren't
 * Sets `gh` up to use TTY in a way that other scripts aren't

Repeating what I wrote in my [comment to Tyler](https://github.com/cli/cli/pull/9783/files#r1806157923):

> But I want to be clear here, I'm not saying that these should be removed. I am saying that I think it's important that there is a clear and shared pattern to all the testscripts that we can say "make it look like that". That can also mean going back to the scripts that are already being merged and making them use this pre and post pattern.

I might be happy with "do whatever you need to do to assert stdout of every exec", I might be happy with "only assert on stdout of non-TTY", and I might be happy with "only assert on stdout of commands that are being used to verify state rather than cause mutations" but I'm not really happy with "all the tests do different things depending on who wrote them" because the knock on effects of that have the potential to be kind of annoying.

So I'm trying to understand how valuable these assertions are to you and to the script itself, and to trade that off against any costs, so that we can tease out the guidelines we want to encourage.

Hope that makes sense.

Reply:
> So I'm trying to understand how valuable these assertions are to you and to the script itself, and to trade that off against any costs, so that we can tease out the guidelines we want to encourage.

I can acquiesce that some of these assertions might not be necessary, or rather that some might not find them necessary.

Personally, I value asserting the context of commands as closed to where they are being made as possible.  It isn't always doable hence sometimes I assert the output of the command while later on trying to use the information and assert that, too.   You could say I care about how the product behaves as well as the state it is changing.  Again, I can acquiesce my personal values for now as I have refactored these tests in the manner of `gh workflow` testscripts, which I feel increases my confidence more.

Reply:
>  but I'm not really happy with "all the tests do different things depending on who wrote them" because the knock on effects of that have the potential to be kind of annoying.

I really agree with this. 

:2c: When I went into writing some of these tests, I was under the impression that interactivity is out of scope. Forcing TTY on feels like we risk stepping over that line, or perhaps risking a command accidentally prompting and hanging the tests for some reason.

I also think that asserting that output looks a certain way is something already covered by our unit tests. 

I'm wondering what asserting on forced TTY output gains us over the assertion that comes with `exec` & the exit code?